### DISCUSSION TITLE
v0.9.4 release

### DISCUSSION THREAD

**Discussion by yifanfeng97** [iMoonLab/DeepHypergraph#48] [Announcements]
We are releasing the v0.9.4 version now! In v0.9.4, we add 6 hypergraph datasets and fix some known bugs. 

New Datasets:
- [DBLP-4k](https://deephypergraph.readthedocs.io/en/0.9.4/generated/dhg.data.DBLP4k.html) @yifanfeng97  
- [IMDB4k](https://deephypergraph.readthedocs.io/en/latest/generated/dhg.data.IMDB4k.html) @yifanfeng97  
- [Recipe100k](https://deephypergraph.readthedocs.io/en/latest/generated/dhg.data.Recipe100k.html) @yifanfeng97 
- [Recipe200k](https://deephypergraph.readthedocs.io/en/latest/generated/dhg.data.Recipe200k.html) @yifanfeng97 
- [Yelp3k](https://deephypergraph.readthedocs.io/en/latest/generated/dhg.data.Yelp3k.html) @yifanfeng97 
- [Tencent2k](https://deephypergraph.readthedocs.io/en/latest/generated/dhg.data.Tencent2k.html) @yifanfeng97 

Fix bugs:
- [fix last layer bn bugs](https://github.com/iMoonLab/DeepHypergraph/commit/27f944f6aa5c66da7f395124ecb72f62c186b53b) @yifanfeng97 
- [fix device error of graph,di_graph,bi_graph,hypergraph laplacian computing](https://github.com/iMoonLab/DeepHypergraph/commit/834634904fed8e0055a6f3e11c616d4a35e7a5f5) @yifanfeng97 

New features:
- [add custom hypergraph generator](https://github.com/iMoonLab/DeepHypergraph/commit/6278f0da6c34e6c40c6f92544c50f010afe757f0) @yifanfeng97 

<hr /><em>This discussion was created from the release <a href='https://github.com/iMoonLab/DeepHypergraph/releases/tag/v0.9.4'>v0.9.4 release</a>.</em>

**Comment by ddl-donglin** [comment]
Why are you so niubility? Orz

**Reply by yifanfeng97** [reply]
You niubility, too. Orz

**Reply by Liker79** [reply]
DaLao，we've recently been working on a project that doesn't involve multimodal data, where we want to use a single object as a vertex, connected by some relationship, which acts as a superedge, and then classify the entire hypergraph. But I see only vertex classification in the documentation, and I wonder if your framework can implement full graph classification. Or can I add a full connection layer after all vertexes output to achieve full graph output?
大佬，我们最近在做相关的课题，不涉及多模态的数据，想以单一的客体做节点，通过某些关系连接起来，这些关系作为超边，然后针对整张超图做分类。但是我看到文档中仅有节点分类的内容，不知道您的框架能否实现全图分类。或者我能否在所有节点输出之后加一个全连接层实现全图输出？

**Comment by 81jar** [comment]
Hi.

I´m installing you later version of dhg (**pip install C:\Users\HP\Desktop\investigacion\energia\DeepHypergraph-0.9.4\DeepHypergraph-0.9.4**) and it present the following error:

ERROR: Could not find a version that satisfies the requirement torch<2.0,>=1.12.1 (from dhg) (from versions: 2.2.0, 2.2.1, 2.2.2, 2.3.0, 2.3.1, 2.4.0, 2.4.1, 2.5.0, 2.5.1)
ERROR: No matching distribution found for torch<2.0,>=1.12.1

And when I try it with "pip install dhg", I have the error: "python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [15 lines of output]
      The 'sklearn' PyPI package is deprecated, use 'scikit-learn'
      rather than 'sklearn' for pip commands."

I need yor help.

**Reply by 81jar** [reply]
I have already installed the scikit-learn package.
My python version is 3.12.5

**Reply by effes3** [reply]
have you solved your problem?
